humility-core/src/core.rs

```diff
@@ -279,23 +282,34 @@ impl Core for ProbeCore {
         self.halt_and_read(|core| Ok(core.read_8(addr, data)?))
     }
 
-    fn read_reg(&mut self, reg: ARMRegister) -> Result<u32> {
+    fn read_reg(&mut self, reg: Register) -> Result<u32> {
         let mut core = self.session.core(0)?;
+        let reg_id = Register::to_u16(&reg).unwrap();
+
         use num_traits::ToPrimitive;
 
         Ok(core.read_core_reg(Into::<probe_rs::CoreRegisterAddress>::into(
-            ARMRegister::to_u16(&reg).unwrap(),
+            reg_id,
         ))?)
     }
 
-    fn write_reg(&mut self, reg: ARMRegister, value: u32) -> Result<()> {
+    fn write_reg(&mut self, reg: Register, value: u32) -> Result<()> {
         let mut core = self.session.core(0)?;
+        let mut reg_id = Register::to_u16(&reg).unwrap();
+
+        //See table 3.3 in the riscv debug spec.
+        //(https://raw.githubusercontent.com/riscv/riscv-debug-spec/master/riscv-debug-stable.pdf#table.3.3)
+        //The pc is read through the DPC register
+        if let Register::RiscV(rv_reg) = reg {
+            reg_id += 0x1000;
+            if rv_reg == RVRegister::PC {
+                reg_id = 0x7b1;
+            }
+        }
```

Comment:
For the ProbeCore, this is actually a bug.  I must have made a bad merge that pulled it in.
However, a similar pattern is used for the OpenOCD Core.  This is due to the different tools representing registers with different numbers.  

Some tools use the[ dwarf register numbers](https://github.com/riscv-non-isa/riscv-elf-psabi-doc/blob/master/riscv-dwarf.adoc) and others use the [debug spec numbers](https://raw.githubusercontent.com/riscv/riscv-debug-spec/master/riscv-debug-stable.pdf#table.3.3).  Additionally, Openocd uses the DWARF numbers but offset the csrs by 65.

Internal to humility I decided to use the debug spec numbers as I think they are the most reasonable assignments, but if anyone has a good reason to switch, I can always change them.

Reply:
I think picking any one of the numbers is just fine. I am just thinking if it would be possible to hide all this translation into the arch specific files i.e. just make a call to convert id in arch code and then use that to perform reg ops. (I posted this comment on another file as well).

Reply:
The latest push refactored a bit to try to move all of the arch specific stuff into their own module.